src/main/rx/fport.c

```diff
@@ -203,10 +207,12 @@ static void fportDataReceive(uint16_t c, void *data)
 
         frameStartAt = currentTimeUs;
         framePosition = 1;
+        rxBuffer[rxBufferWriteIndex].frameStartTimeUs = currentTimeUs;
     } else if (framePosition > 0) {
         if (framePosition >= BUFFER_SIZE + 1) {
                 framePosition = 0;
 
+                rxBuffer[rxBufferWriteIndex].frameStartTimeUs = 0;
```

Comment:
This seems to be wrong - there is no frame received here, and the time the last frame was received does not magically change to the startup of the firmware either.

Reply:
It doesn't really matter as this element gets set when the `FPORT_FRAME_MARKER` character is processed. This is just to reset it to a default state as we've invalidated the frame - either through a timeout or by excessive length.

Reply:
I can not see how a fact (in this case the point in time that an event happened) can be 'invalidated' - if we want this to represent some made up thing then we need to rename it appropriately.

Reply:
It's not really worth arguing about as it has no functional impact. It was just for completeness. Both points you referenced are where the current frame has been invalidated and reset to wait for the next `FPORT_FRAME_MARKER`. When that's received the `frameStartTimeUs` will be reset to the current time when the new frame started.